test/python/compiler/test_assembler.py

```diff
@@ -593,6 +598,343 @@ def test_pulse_gates_delay_only(self):
+            pulse.MeasureChannel(1): 6.64e9,
+            pulse.DriveChannel(4): 4.91e9,
+            pulse.MeasureChannel(3): 6.1e9,
+        }
+        user_lo_config2 = pulse.LoConfig(user_lo_config_dict2)
+        qobj = assemble(
+            [self.circ, self.circ],
+            self.backend,
+            qubit_lo_freq=self.default_qubit_lo_freq,
+            meas_lo_freq=self.default_meas_lo_freq,
+            schedule_los=[self.user_lo_config, user_lo_config2],
+        )
+        validate_qobj_against_schema(qobj)
+
+        qubit_lo_freq_GHz = [freq / 1e9 for freq in self.default_qubit_lo_freq]
+        meas_lo_freq_GHz = [freq / 1e9 for freq in self.default_meas_lo_freq]
+        self.assertListEqual(qobj.config.qubit_lo_freq, qubit_lo_freq_GHz)
+        self.assertListEqual(qobj.config.meas_lo_freq, meas_lo_freq_GHz)
+        self.assertEqual(len(qobj.experiments), 2)
+
+        # expt 0 los
+        self.assertEqual(qobj.experiments[0].config.qubit_lo_freq, [5.55, 5, 5, 4.91, 5])
+        self.assertEqual(
+            qobj.experiments[0].config.meas_lo_freq, [6.64, 6.7, 6.7, 6.7, 6.1]
+        )
+        # expt 1 los
+        self.assertEqual(qobj.experiments[1].config.qubit_lo_freq, [5, 5.55, 5, 5, 4.91])
+        self.assertEqual(
+            qobj.experiments[1].config.meas_lo_freq, [6.7, 6.64, 6.7, 6.1, 6.7]
+        )
+
+    def test_assemble_multi_circ_single_lo_config(self):
+        """Test assembling multiple circuits, with a single expt level lo config (should override
+        job level)."""
+        qobj = assemble(
+            [self.circ, self.circ],
+            self.backend,
+            qubit_lo_freq=self.default_qubit_lo_freq,
+            meas_lo_freq=self.default_meas_lo_freq,
+            schedule_los=self.user_lo_config,
+        )
+        validate_qobj_against_schema(qobj)
+
+        self.assertListEqual(qobj.config.qubit_lo_freq, [5.55, 5, 5, 4.91, 5])
+        self.assertListEqual(qobj.config.meas_lo_freq, [6.64, 6.7, 6.7, 6.7, 6.1])
+        self.assertEqual(len(qobj.experiments), 2)
+
+    def test_assemble_multi_circ_wrong_number_of_multi_lo_configs(self):
+        """Test assembling circuits, with a different number of expt level lo configs (n:m setup).
+        """
+        with self.assertRaises(QiskitError):
+            assemble(
+                [self.circ, self.circ, self.circ],
+                self.backend,
+                qubit_lo_freq=self.default_qubit_lo_freq,
+                meas_lo_freq=self.default_meas_lo_freq,
+                schedule_los=[self.user_lo_config, self.user_lo_config],
+            )
+
+    def test_assemble_circ_lo_config_errors(self):
+        """Test that lo config errors are raised properly if expt level los are provided and some
+        are missing or if default values are not provided. Also check that expt level lo range
+        is validated."""
+        # no defaults, but have drive/meas expt level los for each qubit (no error)
```

Comment:
```suggestion
        # no defaults, but have drive/measure experiment level LOs for each qubit (no error)
```

Reply:
Fix `expt` elsewhere please.

Reply:
Addressed in https://github.com/Qiskit/qiskit-terra/pull/6167/commits/705ca3c97ae84c6f950f430f1eea576d4eb93e1c.